pal/src/thread/pal_thread.cpp

```diff
@@ -191,6 +191,7 @@ static void InternalEndCurrentThreadWrapper(void *arg)
 --*/
 BOOL TLSInitialize()
 {
+    free_threads_spinlock.Reset();
```

Comment:
This will be called once and can't be called during/after another lock in place due to `PAL` is not yet initialized. The underlying issue here is related to whole lib/pal initialization on start. I did some improvements there in the past, there will be more to come.